README.md

```diff
@@ -434,6 +439,116 @@ status:
+
+If an Application Resource Mapping defines `containers`, it **MUST NOT** define `.envs` and `.volumeMounts`.  If an Application resources does not define `containers`, it **MUST** define `.envs` and `.volumeMounts`.
+
+[carm-crd]: service.binding_clusterapplicationresourcemappings.yaml
+[container]: https://kubernetes.io/docs/reference/kubernetes-api/workloads-resources/container/
+[crd-syntax]: https://kubernetes.io/docs/tasks/extend-kubernetes/custom-resources/custom-resource-definitions/#create-a-customresourcedefinition
+[envvar]: https://kubernetes.io/docs/reference/kubernetes-api/workloads-resources/container/#environment-variables
+[jsonpath]: https://kubernetes.io/docs/reference/kubectl/jsonpath/
+[volume]: https://kubernetes.io/docs/reference/kubernetes-api/config-and-storage-resources/volume
+[volumemount]: https://kubernetes.io/docs/reference/kubernetes-api/workloads-resources/container/#volumes
+
+### Resource Type Schema
+
+```yaml
+apiVersion: service.binding/v1alpha2
+kind: ClusterApplicationResourceMapping
+metadata:
+  name:                 # string
+  generation:           # int64, defined by the Kubernetes control plane
+  ...
+spec:
+  versions:             # []Version
+  - version:            # string
+    containers:         # []string, optional
+    envs:               # []string, optional
+    volumeMounts:       # []string, optional
+    volumes:            # string
+```
+
+### Container-based Example Resource
+
+```yaml
+apiVersion: service.binding/v1alpha2
+kind: ClusterApplicationResourceMapping
+metadata:
+ name:  cronjobs.batch
+spec:
+  versions:
+  - version: v1beta1
+    containers:
+    - .spec.jobTemplate.spec.template.spec.containers
+    - .spec.jobTemplate.spec.template.spec.initContainers
+    volumes: .spec.jobTemplate.spec.template.spec.volumes
+  - version: v2alpha1
+    containers:
+    - .spec.jobTemplate.spec.template.spec.containers
+    - .spec.jobTemplate.spec.template.spec.initContainers
+    volumes: .spec.jobTemplate.spec.template.spec.volumes
```

Comment:
including both versions here instead of the wildcard has me trying to hunt for the difference between the two (they are identical other than the version). If we want to have an example that is doesn't use the version wildcard, then I'd suggest keeping just `v1beta1`.

```suggestion
  - version: "*"
    containers:
    - .spec.jobTemplate.spec.template.spec.containers
    - .spec.jobTemplate.spec.template.spec.initContainers
    volumes: .spec.jobTemplate.spec.template.spec.volumes
```